### PR TITLE
Remove api test case

### PR DESCRIPTION
**Pull request by mshriver**
Now that all tests have been converted to pytest, we no longer have a need for the APITestCase class.

This PR removes that class, and changes the upgrade test setup pattern so that it no longer relies on the base TestCase class that was based on UnitTest. 

As a result of removing those base classes, I could also remove the assertion classes and unit tests related to behavior of those base classes. All of that behavior is now handled natively by pytest.

# Test Results

## test_permission.py

```
setup@localhost:~/repos/robottelo (remove-APITestCase%<>) % pytest -x tests/foreman/api/test_permission.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, reportportal-5.0.8, mock-3.6.0, cov-2.11.1, services-2.2.1
collected 15 items                                                                                                                                                                           

tests/foreman/api/test_permission.py ...............
```

## test_organization.py

The failing test fails in master branch as well.

```
setup@localhost:~/repos/robottelo (remove-APITestCase*%<>) % pytest tests/foreman/api/test_organization.py 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, reportportal-5.0.8, mock-3.6.0, cov-2.11.1, services-2.2.1
collected 43 items                                                                                                                                                                           

tests/foreman/api/test_organization.py F.......................................... 
```

## test_subscription.py

`tests.foreman.api.test_subscription.SubscriptionsTestCase.test_sca_end_to_end` fails in master branch with the same exception.

```
setup@localhost:~/repos/robottelo (remove-APITestCase*%<>) % pytest tests/foreman/api/test_subscription.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, reportportal-5.0.8, mock-3.6.0, cov-2.11.1, services-2.2.1
collected 9 items                                                                                                                                                                            

tests/foreman/api/test_subscription.py .......F. 
```

## test_user.py

Test failures are equivalent to or better than master branch.

```
setup@localhost:~/repos/robottelo (remove-APITestCase%<>) % pytest --tb=line tests/foreman/api/test_user.py
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, reportportal-5.0.8, mock-3.6.0, cov-2.11.1, services-2.2.1
collected 141 items                                                                                                                                                                          

tests/foreman/api/test_user.py ...................................................................................................................................EEEEEEEE.F           [100%]
```

## upgrades/test_host.py

This is failing in master with a locator error, I'm not going to address that here.

### CODE REVIEW

File: tests/foreman/api/test_user.py [hunk 1]
```diff
@@ -44,14 +44,15 @@
 from robottelo.helpers import read_data_file
 
 
+@pytest.fixture(scope='module')
+def create_user(self):
+    """Create an user"""
```
Comment (JacobCallahan) [inline tests/foreman/api/test_user.py:49]:
```suggestion
    """Create a user"""
```

File: tests/robottelo/cli/test_assertions.py [hunk 2]
```diff
@@ -3,7 +3,28 @@
 from robottelo.cli.assertions import assert_cli_not_raises
 from robottelo.cli.assertions import assert_cli_not_raises_regex
 from robottelo.cli.base import CLIReturnCodeError
-from tests.robottelo.test_assertions import fake_128_return_code
+
+
+def fake_128_return_code():
+    """Fake CLI response with 128 return_code"""
+    # flake8:noqa (line-too-long)
```
Comment (JacobCallahan) [inline tests/robottelo/cli/test_assertions.py:10]: Do we still need this?

File: tests/foreman/api/test_subscription.py [hunk 3]
```diff
@@ -31,273 +31,280 @@
...
+    :expectedresults: subscription status is 'Disabled'
+
+    :BZ: 1789924
+
+    :CaseImportance: Medium
+    """
+    rhel7_contenthost.install_katello_ca()
+    rhel7_contenthost.register_contenthost(module_org.label, module_ak.name)
+    assert rhel7_contenthost.subscribed
+    host = entities.Host().search(query={'search': f'name={rhel7_contenthost.hostname}'})
```
Comment (JacobCallahan) [inline tests/foreman/api/test_subscription.py:250]:
```suggestion
    host = rhel7_contenthost.nailgun_host
```

File: tests/foreman/api/test_subscription.py [hunk 4]
```diff
@@ -31,273 +31,280 @@
...
+    assert entities.Organization(id=module_org.id).read().simple_content_access is True
+    # Verify that you cannot attach a subscription to an activation key in SCA Mode
+    subscription = entities.Subscription(organization=module_org).search(
+        query={'search': f'name="{DEFAULT_SUBSCRIPTION_NAME}"'}
+    )[0]
+    with pytest.raises(HTTPError) as ak_context:
+        module_ak.add_subscriptions(data={'quantity': 1, 'subscription_id': subscription.id})
+    assert 'Simple Content Access' in ak_context.value.response.text
+    # Verify that you cannot attach a subscription to an Host in SCA Mode
+    host_id = entities.Host().search(query={'search': rhel7_contenthost.hostname})[0].id
```
Comment (JacobCallahan) [inline tests/foreman/api/test_subscription.py:282]:
```suggestion
    host_id = rhel7_contenthost.nailgun_host.id
```

File: tests/foreman/api/test_subscription.py [hunk 5]
```diff
@@ -31,273 +31,280 @@
...
-        package = self.content_host.run('yum install -y python-pulp-manifest')
-        assert 'Complete!' in package.stdout or 'already installed' in package.stdout
+    assert 'Simple Content Access' in host_context.value.response.text
+    # Create a content view with repos and check to see that the client has access
+    content_view = entities.ContentView(organization=module_org).create()
+    content_view.repository = [rh_repo, custom_repo]
+    content_view.update(['repository'])
+    content_view.publish()
+    assert len(content_view.repository) == 2
+    host = entities.Host().search(query={'search': rhel7_contenthost.hostname})[0]
```
Comment (JacobCallahan) [inline tests/foreman/api/test_subscription.py:294]:
```suggestion
    host = rhel7_contenthost.nailgun_host
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_user.py (hunk 1 above)
Comment (mshriver) [reply tests/foreman/api/test_user.py:49]: At least Stephen didn't catch me doing this

File: tests/robottelo/cli/test_assertions.py (hunk 2 above)
Comment (mshriver) [reply tests/robottelo/cli/test_assertions.py]: The whole module is gone now :dancers:

File: tests/foreman/api/test_subscription.py (hunk 3 above)
Comment (mshriver) [reply tests/foreman/api/test_subscription.py:250]: golfed this down even further, using that new property though.

File: tests/robottelo/cli/test_assertions.py (hunk 2 above)
Comment (JacobCallahan) [reply tests/robottelo/cli/test_assertions.py]: that's one way to do it!

File: tests/foreman/api/test_organization.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 """Unit tests for the ``organizations`` paths.
 
-Each ``APITestCase`` subclass tests a single URL. A full list of URLs to be
+Each class tests a single URL. A full list of URLs to be
 tested can be found here:
```
Comment (swadeley) [inline tests/foreman/api/test_organization.py:4]:
```suggestion
tested can be found on your Satellite:
```

File: tests/foreman/api/test_organization.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
 http://theforeman.org/api/apidoc/v2/organizations.html
```
Comment (swadeley) [inline tests/foreman/api/test_organization.py:5]:
```suggestion
http://<sat6>/apidoc/v2/organizations.html
```

File: tests/foreman/api/test_permission.py [hunk 8]
```diff
@@ -1,6 +1,6 @@
 """Unit tests for the ``permissions`` paths.
 
-Each ``APITestCase`` subclass tests a single URL. A full list of URLs to be
+Each class tests a single URL. A full list of URLs to be
 tested can be found here: http://theforeman.org/api/apidoc/v2/permissions.html
```
Comment (swadeley) [inline tests/foreman/api/test_permission.py:4]:
```suggestion
tested can be found on your Satellite: http://<sat6>/apidoc/v2/permissions.html
```

File: tests/foreman/api/test_user.py [hunk 9]
```diff
@@ -1,6 +1,6 @@
 """Unit tests for the ``users`` paths.
 
-Each ``APITestCase`` subclass tests a single URL. A full list of URLs to be
+Each class tests a single URL. A full list of URLs to be
 tested can be found here: http://theforeman.org/api/apidoc/v2/users.html
```
Comment (swadeley) [inline tests/foreman/api/test_user.py:4]:
```suggestion
tested can be found on your Satellite: http://<sat6>/apidoc//v2/users.html
```

**Review by swadeley — APPROVED** [review]
ACK (with suggestions, because the links to upstream API docs no longer work, and in any case we should consider the Satellite under test)

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**